### DISCUSSION TITLE
Xtion Subscribe Error after 22.04 clean install at HSR-B

### DISCUSSION THREAD

**Discussion by porizou** [HumanSupportRobot/QandA#31] [Q&A]
After a clean install of HSRB to 22.04, the RGBD camera is no longer available.
The image is using 'hsrb-installer_22.04.01_focal_HSRB-PHASE5_ja_JP.UTF-8'

![Screenshot from 2022-12-07 14-11-36](https://user-images.githubusercontent.com/14184141/206095294-88f6344e-aa08-4ed9-aabd-37299583ef63.png)

The Xtion itself is recognized because ASUS is visible with the lsusb command.

**Comment by hsr-admin** [comment]
The following URL may be helpful in resolving the issue.
In particular, I have not heard of any software problems related to Xtion in 22.04.
https://docs.hsr.io/hsr_log_analysis_manual_en/log_files.html#id1
https://docs.hsr.io/hsrb_maintenance_manual_en/src/check_xtion.html

**Comment by adamgolding1** [comment]
Has there been a solution to this issue? We are also having this problem. We did a clean install of 22.04 and did an upgrade to 22.10 and still have this issue. The master.log file is below.
[master-afterclean.log](https://github.com/HumanSupportRobot/QandA/files/11046603/master-afterclean.log)

**Reply by hsr-admin** [reply]
Have you been able to recover it using the following?
https://docs.hsr.io/hsrb_maintenance_manual_en/src/check_xtion.html

**Reply by adamgolding1** [reply]
Yes I have tried this procedure. In fact I have been able to get the Xtion to run correctly sometimes for 10 minutes but then it shuts down randomly. After this occurs it only ever comes back online for maybe 1 minute before crashing again. This has all been performed on a clean install robot with no other software running. Any help would be very much appreciated.

**Reply by hsr-admin** [reply]
I heard that there is a case of improvement by putting a USB hub in between. Please wait a moment while we check.

**Reply by hsr-admin** [reply]
The cases we identified that were recovered were hand cameras and stereo cameras.

**Reply by hsr-admin** [reply]
I wonder if we need to identify where the problem lies below?

- Xtion: Check if it works by itself.
- USB cable: Replace with an external cable to see if it works.
- SUGOI hub: Replace the hub and see if it works.

If we can identify some of the defects, we may be able to take some effective measures.

(Reference)
It is unknown if it is possible to purchase from overseas, but "SUGOI hub" is available at the following website.
https://search.rakuten.co.jp/search/mall/%E3%82%B9%E3%82%B4%E3%82%A4%E3%83%8F%E3%83%96/

**Reply by adamgolding1** [reply]
Thank you for your feedback, I will start to try these options and update you shortly.

**Reply by adamgolding1** [reply]
I tried the following:
Xtion: Check if it works by itself 
- Yes I connected directly to laptop and works well without shutting down
USB cable: Replace with an external cable to see if it works 
- I connected an external cable to the SUGOI hub and still not too stable.  
SUGOI hub: Replace the hub and see if it works
- I couldn't find the SUGOI hub in Australia, is there a recommended alternative?

Is it possible to bypass the SUGOI hub and connect the internal cable to the motherboard to test as well? What is the purpose of the SUGOI hub?

Thankyou so much for your help so far.

**Reply by hsr-admin** [reply]
In one case, the problem was solved by replacing the hubs on the sugoi with new ones.

Please wait a moment as we check to see if there is a recommended alternative to the SUGOI hub or if there are other means available.

**Reply by adamgolding1** [reply]
Update - I tried to replace the SUGOI hub with a similar powered USB hub and so far the HSR seems to be stable and runnng without errors.

**Reply by hsr-admin** [reply]
Glad to hear it's running stable.
I have heard that preparations are underway to transport the SUGOI hub from Japan to you.

**Reply by adamgolding1** [reply]
Thank you so much for you help with everything

**Reply by hsr-admin** [reply]
We would like to cancel the transportation of the SUGOI HUB because the replacement HUB is working without any problems.
Please understand that the cost of transporting the HUB and the procedures require man-hours.
Also, could you please provide us with the product information of the HUB you replaced for future reference?

**Reply by porizou** [reply]
The problem was not a software problem, but a hardware problem,
I replaced the SUGOI hub with this new one and it works fine.
https://www.yodobashi.com/product/100000001001083777/

**Reply by adamgolding1** [reply]
We used this generic powered usb 2.0 hub, I am going to try a newer powered usb 3.0 hub soon and will let you know the outcome. 
![IMG_0229](https://user-images.githubusercontent.com/116816170/233248208-9941c754-3e5a-4b41-b824-dafa656dedf1.JPG)
![IMG_0230](https://user-images.githubusercontent.com/116816170/233248255-c2af2d9e-31dd-45e6-9952-d1ebfcf91d31.JPG)
